joint_limits/test/test_simple_joint_limiter.hpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2023, Stogl Robotics Consulting UG (haftungsbeschränkt)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TEST_SIMPLE_JOINT_LIMITER_HPP_
+#define TEST_SIMPLE_JOINT_LIMITER_HPP_
+
+#include <gmock/gmock.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+#include "joint_limits/joint_limiter_interface.hpp"
+#include "joint_limits/joint_limits.hpp"
+#include "pluginlib/class_loader.hpp"
+#include "rclcpp/duration.hpp"
+#include "rclcpp/node.hpp"
+#include "trajectory_msgs/msg/joint_trajectory_point.hpp"
+
+const double COMMON_THRESHOLD = 0.0011;
+
+#define CHECK_STATE_SINGLE_JOINT(tested_traj_point, idx, expected_pos, expected_vel, expected_acc) \
+  EXPECT_NEAR(tested_traj_point.positions[idx], expected_pos, COMMON_THRESHOLD);                   \
+  EXPECT_NEAR(tested_traj_point.velocities[idx], expected_vel, COMMON_THRESHOLD);                  \
+  EXPECT_NEAR(tested_traj_point.accelerations[idx], expected_acc, COMMON_THRESHOLD);
+
+#define INTEGRATE(cur, des, dt)                                                      \
+  cur.positions[0] += des.velocities[0] * dt + 0.5 * des.accelerations[0] * dt * dt; \
+  cur.velocities[0] += des.accelerations[0] * dt;
```

Comment:
We prefer to use private methods instead of defines.

Also, a good practice that we figured over time is when having this in methods to add additional output for easier debugging of tests when needed.

Check here the example: https://github.com/ros-controls/ros2_control/blob/master/controller_manager/test/test_hardware_management_srvs.cpp#L95-L96

Reply:
> Also, a good practice that we figured over time is when having this in methods to add additional output for easier debugging of tests when needed.

I found that the displayed default text was quite verbose already see :
```
The difference between desired_joint_states_.velocities[0] and 0.635 is 0.010000000000000009, which exceeds COMMON_THRESHOLD, where
2: desired_joint_states_.velocities[0] evaluates to 0.625,
2: 0.635 evaluates to 0.63500000000000001, and
2: COMMON_THRESHOLD evaluates to 0.0011000000000000001.
```

the question is then, is the good practice to make the output simpler clearer **without** too much output ?
for instance such an ASSERT should just write in a compact form "Desired velocity (0.625) exceeds expected (0.635) by (0.01 > 0.0011)" is that what you meant ? 
I think the example you pointed at adds "semantic" and "context" to the values tested. 
I roughly grasp the idea but would like to know if good practice is to have concise output with context (requires reformatting existing output) or long output with context (there might be a way to add context only to the output without changing existing formatting of the verbose error)

Reply:
You are right, we can leave it as it is - using MACRO